backend/apps/openai/main.py

```diff
@@ -234,72 +228,66 @@ def merge_models_lists(model_lists):
     return merged_list
 
 
-async def get_all_models(raw: bool = False):
+def is_openai_api_disabled():
+    api_keys = app.state.config.OPENAI_API_KEYS
+    no_keys = len(api_keys) == 1 and api_keys[0] == ""
+    return no_keys or not app.state.config.ENABLE_OPENAI_API
+
+
+async def get_all_models_raw() -> list:
```

Comment:
Not a huge fan of having a separate `get_all_models_raw` function, perhaps it's better to make this inner function for `get_all_models` to enable `raw=True` param?

Reply:
LSP's don't like it when the output type of a function depends on an argument: at least PyRight isn't smart enough to figure out that `get_all_models()` returns a list if and only if `raw` is set to `True`, and starts yapping about how you can't use a `list` as a `dict` (when raw is False)

Only other way to make the warnings go away is to call `assert isinstance` after each call to `get_all_models()` or implement some monstrosity with generics.

My hope is that eventually we can add a static type checker to core backend components. Makes it a lot easier to catch errors early.

Reply:
I wouldn't want to change the current function just for LSP's sake. Perhaps`Union[dict, list]` could work?

Reply:
Found a solution with `@overload`.